docs/serving/spec/knative-api-specification-1.0.md

```diff
@@ -148,6 +148,39 @@ Service object to reduce the cognitive overhead for application developers – t
 Service object should provide sufficient controls to cover most of application
 deployment scenarios (by frequency).
 
+## Extentions
+
+Extending the Knative resource model allows for custom semantics to be
+offered by implementions of the specification. Unless otherwise noted,
+implementations of this specification MAY define extensions but those
+extensions MUST NOT contradict the semantics defined within this specification.
+
+There are several ways in which implementations can extend the model:
+- Annotations and Labels
+  Allowing end users to include annotations or labels on the Knative resources
+  allows for them to indicate that they would like some additional semantics
+  applied to those resources. When defining annotations, or labels, it
+  is STRONGLY RECOMMENDED that they have some vendor-specific prefix to
+  avoid any potential naming conflict with other extensions or future
+  annotations defined by the specification.
+
+  Note that this would allow for new controllers to be added to the system,
+  to detect these annotations/labels and act upon them, without requiring
+  code changes to the core Knative components. As such, when possible, this
+  mechanism is preferred.
```

Comment:
Maybe put this first, so it's clear what the recommendation is?

Reply:
Not following, this *is* the first one.  Can you elaborate?

Reply:
I meant move this paragraph before most of the other one.

```md
   _NOTE_: Because this mechanism allows new controllers to be added to the
   system without requiring code changes to the core Knative components, it is
   the preferred mechanism for extending the Knative interface.

   Allowing end users to include annotations or labels on the Knative resources
   allows for them to indicate that they would like some additional semantics
   applied to those resources...
```

Reply:
Done